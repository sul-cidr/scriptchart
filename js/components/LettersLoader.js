import React from "react";

import LetterButton from "./LetterButton";

import SyriacLetter, { letters } from "./SyriacLetter";

class LettersLoader extends React.Component {
  constructor(props) {
    super(props);
    this.state = { letterButtons: [],
                   selectedLetters: [] };

    this.buttonClick = this.buttonClick.bind(this);
  }

  buttonClick(letterID, operation) {
    let selectedLetters = [...this.state.selectedLetters];
    if (operation == "select") {
      selectedLetters.push(letters.find(lt => lt['id'] == letterID));
    } else {
      selectedLetters.splice(selectedLetters.findIndex(lt => lt['id'] == letterID), 1);
    }
    this.props.handleSelect("letters", selectedLetters);
    this.setState({selectedLetters});
  }

  // If the API data source is not available, the set of letters to be
  // displayed will be the IDs of the active letters data structure
  // in SyriacLetter.js
  //data = Object.keys(SyriacLetter.letterInfo);

  componentDidMount() {
    fetch("http://localhost:8000/api/letters?format=json")
      .then(response => {
        //return response.json();
        // XXX Probably need to update the DB with the data from
        // letterInfo, or just not use the DB as source letter data
        return letters;
      })
      /* In production, it's likely preferable for the menu to display
       * a blank list when the backend API is down, rather than
       * displaying data that may not reflect the database state.
       */
      .catch(function(error) {
        return letters;
      })
      .then(data => {
        let buttons = data.map(lt => {
          return (
            <LetterButton key={lt.id} letterID={lt.id} onLetterClick={this.buttonClick}
              letter={<SyriacLetter id={lt.id} />}
            />
          );
        });
        this.setState({ letterButtons: buttons });
      });
  }
  render() {
    return (
      <div className={"buttons are-small"}>{this.state.letterButtons}</div>
    );
  }
}

export default LettersLoader;
