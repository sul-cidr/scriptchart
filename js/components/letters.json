[
  {
    "id": 2,
    "letter": "Alaph (Angular)",
    "is_script": true,
    "display": "ܐ",
    "script": "estrangela"
  },
  {
    "id": 4,
    "letter": "Alaph (Round)",
    "is_script": true,
    "display": "ܐ",
    "script": "serto"
  },
  {
    "id": 9,
    "letter": "Beth",
    "is_script": true,
    "display": "ܒ",
    "script": "estrangela"
  },
  {
    "id": 14,
    "letter": "Gamal",
    "is_script": true,
    "display": "ܓ",
    "script": "estrangela"
  },
  {
    "id": 12,
    "letter": "Dalath (Angular)",
    "is_script": true,
    "display": "ܕ",
    "script": "estrangela"
  },
  {
    "id": 13,
    "letter": "Dalath (Round)",
    "is_script": true,
    "display": "ܕ",
    "script": "serto"
  },
  {
    "id": 16,
    "letter": "He (Angular)",
    "is_script": true,
    "display": "ܗ",
    "script": "estrangela"
  },
  {
    "id": 17,
    "letter": "He (Round)",
    "is_script": true,
    "display": "ܗ",
    "script": "serto"
  },
  {
    "id": 46,
    "letter": "Waw",
    "is_script": true,
    "display": "ܘ",
    "script": "estrangela"
  },
  {
    "id": 50,
    "letter": "Zain",
    "is_script": true,
    "display": "ܙ",
    "script": "estrangela"
  },
  {
    "id": 18,
    "letter": "Heth",
    "is_script": true,
    "display": "ܚ",
    "script": "estrangela"
  },
  {
    "id": 45,
    "letter": "Teth",
    "is_script": true,
    "display": "ܛ",
    "script": "estrangela"
  },
  {
    "id": 48,
    "letter": "Yudh (Connected)",
    "is_script": true,
    "display": "ܝ",
    "script": "estrangela",
    "leading_letter": "ܝ",
    "trailing_letter": "ܝ"
  },
  {
    "id": 49,
    "letter": "Yudh (Stand-alone)",
    "is_script": true,
    "display": "ܝ",
    "script": "estrangela"
  },
  {
    "id": 19,
    "letter": "Kaph",
    "is_script": true,
    "display": "ܟ",
    "script": "estrangela",
    "leading_letter": "ܝ"
  },
  {
    "id": 20,
    "letter": "Kaph (Final)",
    "is_script": true,
    "display": "ܟ",
    "script": "estrangela"
  },
  {
    "id": 25,
    "letter": "Lamadh (Final, open)",
    "is_script": true,
    "display": "ܠ",
    "script": "estrangela"
  },
  {
    "id": 21,
    "letter": "Lamadh",
    "is_script": true,
    "display": "ܠ",
    "script": "estrangela",
    "trailing_letter": "ܝ"
  },
  {
    "id": 23,
    "letter": "Lamadh (Final, closed)",
    "is_script": true,
    "display": "ܠ",
    "script": "serto"
  },
  {
    "id": 26,
    "letter": "Mim",
    "is_script": true,
    "display": "ܡ",
    "script": "estrangela",
    "leading_letter": "ܝ"
  },
  {
    "id": 27,
    "letter": "Mim (Final)",
    "is_script": true,
    "display": "ܡ",
    "script": "estrangela"
  },
  {
    "id": 28,
    "letter": "Nun",
    "is_script": true,
    "display": "ܢ",
    "script": "estrangela",
    "leading_letter": "ܝ"
  },
  {
    "id": 30,
    "letter": "Nun (Final, connected)",
    "is_script": true,
    "display": "ܢ",
    "script": "estrangela",
    "trailing_letter": "ܝ"
  },
  {
    "id": 31,
    "letter": "Nun (Final, unconnected)",
    "is_script": true,
    "display": "ܢ",
    "script": "estrangela"
  },
  {
    "id": 38,
    "letter": "Semkath",
    "is_script": true,
    "display": "ܣ",
    "script": "estrangela"
  },
  {
    "id": 5,
    "letter": "Ayin",
    "is_script": true,
    "display": "ܥ",
    "script": "estrangela"
  },
  {
    "id": 32,
    "letter": "Pe",
    "is_script": true,
    "display": "ܦ",
    "script": "estrangela"
  },
  {
    "id": 37,
    "letter": "Sadhe",
    "is_script": true,
    "display": "ܨ",
    "script": "estrangela"
  },
  {
    "id": 33,
    "letter": "Qaph",
    "is_script": true,
    "display": "ܩ",
    "script": "estrangela"
  },
  {
    "id": 35,
    "letter": "Rish (Angular)",
    "is_script": true,
    "display": "ܪ",
    "script": "estrangela"
  },
  {
    "id": 36,
    "letter": "Rish (Round)",
    "is_script": true,
    "display": "ܪ",
    "script": "serto"
  },
  {
    "id": 39,
    "letter": "Shin",
    "is_script": true,
    "display": "ܫ",
    "script": "estrangela"
  },
  {
    "id": 43,
    "letter": "Taw (Looped)",
    "is_script": true,
    "display": "ܬ",
    "script": "estrangela"
  },
  {
    "id": 44,
    "letter": "Taw (Triangular)",
    "is_script": true,
    "display": "ܬ",
    "script": "serto",
    "trailing_letter": "ܝ"
  },
  {
    "id": 42,
    "letter": "Taw (L-shaped)",
    "is_script": true,
    "display": "ܬ",
    "script": "serto"
  }
]
