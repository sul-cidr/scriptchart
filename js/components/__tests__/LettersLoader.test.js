/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable no-undef */
import React from "react";
import renderer from "react-test-renderer";
import LettersLoader from "../LettersLoader";

import letters from "../letters.json";

describe("LettersLoader test", () => {
  it("LettersLoader should match snapshot", () => {
    const component = renderer.create(
      <LettersLoader selectedLetters={letters.slice(0, 2)} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
