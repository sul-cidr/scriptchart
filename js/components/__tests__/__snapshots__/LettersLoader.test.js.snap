// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LettersLoader test LettersLoader should match snapshot 1`] = `
<div
  className="buttons are-small"
>
  <span
    className="button is-success"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܐ"
    >
      
      ܐ
      
    </span>
  </span>
  <span
    className="button is-success"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "SertoJerusalem",
          "fontSize": "2em",
        }
      }
      title="ܐ"
    >
      
      ܐ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܒ"
    >
      
      ܒ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܓ"
    >
      
      ܓ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܕ"
    >
      
      ܕ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "SertoJerusalem",
          "fontSize": "2em",
        }
      }
      title="ܕ"
    >
      
      ܕ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܗ"
    >
      
      ܗ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "SertoJerusalem",
          "fontSize": "2em",
        }
      }
      title="ܗ"
    >
      
      ܗ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܘ"
    >
      
      ܘ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܙ"
    >
      
      ܙ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܚ"
    >
      
      ܚ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܛ"
    >
      
      ܛ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܝ"
    >
      ‍
      ܝ
      ‍
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܝ"
    >
      
      ܝ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܟ"
    >
      
      ܟ
      ‍
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܟ"
    >
      
      ܟ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܠ"
    >
      
      ܠ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܠ"
    >
      ‍
      ܠ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "SertoJerusalem",
          "fontSize": "2em",
        }
      }
      title="ܠ"
    >
      
      ܠ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܡ"
    >
      
      ܡ
      ‍
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܡ"
    >
      
      ܡ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܢ"
    >
      
      ܢ
      ‍
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܢ"
    >
      ‍
      ܢ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܢ"
    >
      
      ܢ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܣ"
    >
      
      ܣ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܥ"
    >
      
      ܥ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܦ"
    >
      
      ܦ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܨ"
    >
      
      ܨ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܩ"
    >
      
      ܩ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܪ"
    >
      
      ܪ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "SertoJerusalem",
          "fontSize": "2em",
        }
      }
      title="ܪ"
    >
      
      ܪ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܫ"
    >
      
      ܫ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "EstrangeloEdessa",
          "fontSize": "2em",
        }
      }
      title="ܬ"
    >
      
      ܬ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "SertoJerusalem",
          "fontSize": "2em",
        }
      }
      title="ܬ"
    >
      ‍
      ܬ
      
    </span>
  </span>
  <span
    className="button is-outlined"
    onClick={[Function]}
  >
    <span
      style={
        Object {
          "direction": "rtl",
          "fontFamily": "SertoJerusalem",
          "fontSize": "2em",
        }
      }
      title="ܬ"
    >
      
      ܬ
      
    </span>
  </span>
</div>
`;
